### DISCUSSION TITLE
Jaeger Install with Jaeger Operator and with Cassandra API

### DISCUSSION THREAD

**Discussion by jay-annapureddy** [jaegertracing/jaeger-operator#2174] [Q&A]
Can some one help me with the right configuration of Cassandra API of Cosmos DB to be used while installing Jaeger using Jaeger Operator for kubernetes ?

I am using the following but getting timeout error

Connection error: ('Unable to connect to any servers', {'20.61.97.0:10350': OperationTimedOut('errors=Timed out creating connection (5 seconds), last_host=None')})
Cassandra is still not up at jaegertracing.cassandra.cosmos.azure.com:10350. Waiting 1 second.
Checking if Cassandra is up at jaegertracing.cassandra.cosmos.azure.com:10350.

![jaeger-config](https://user-images.githubusercontent.com/109361008/216951011-a1ecd94f-b200-4081-8ccb-2096f43c2862.PNG)

**Comment by jay-annapureddy** [comment]
@yurishkuro @sagaranand015  could you please help providing right example configuration?

**Comment by jay-annapureddy** [comment]
@pavolloffay any help here ?

**Comment by jay-annapureddy** [comment]
Hello All, @yurishkuro, @pavolloffay

I finally figured out what is an issue here, it is with create schema parameter, your help here is much appreciated.

With the following CRD I am able to connect to **Cassandra API of Azure Cosmos DB** but when I access the Jaeger UI it shows error "keyspace is not exists" and when I retry after manually creating keyspace then it shows "service-names table is missing".

Any Idea how can I force jaeger operator to create keyspace and schema for an external **Cassandra API of Azure Cosmos DB** ? 
if no option then is there a way to execute [create.sh ](https://github.com/jaegertracing/jaeger/blob/main/plugin/storage/cassandra/schema/create.sh) script on Cassandra api data explorer from Azure portal ?




<img width="452" alt="jaeger-install-create-schema-error" src="https://user-images.githubusercontent.com/109361008/218194156-3d106d85-51b1-44e9-a92f-5e579014d63e.png">

**Reply by yurishkuro** [reply]
I don't have an answer for CRD. You can always build the schema manually, the `create.sh` script you mentioned simply outputs the scheme content, and if you use our schema-generator Docker image it just pipes that schema content into `| cqlsh`. I assume Azure portal should give you similar capability to access the keyspace via cqlsh.

**Comment by jay-annapureddy** [comment]
Thank you @yurishkuro 

I cloned the repo and able to generate the schema script, I will see how I can inject into my Cassandra API of Azure Cosmos DB

PS C:\Users\jayan\Documents\workspace\jaeger> $env:MODE="test"
PS C:\Users\jayan\Documents\workspace\jaeger> $env:DATACENTER="sbox"
PS C:\Users\jayan\Documents\workspace\jaeger> sh ./plugin/storage/cassandra/schema/create.sh
